azurerm/resource_arm_monitor_diagnostics.go

```diff
@@ -0,0 +1,284 @@
+		},
+		name)
+	if err != nil {
+		return fmt.Errorf("Error creating Diagnostics Setting %q (Resource ID %q): %+v", name, targetResourceId, err)
+	}
+
+	read, err := client.Get(ctx, targetResourceId, name)
+	if err != nil {
+		return err
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read ID for Monitor Diagnostics %q", name)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmMonitorDiagnosticsRead(d, meta)
+}
+
+func resourceArmMonitorDiagnosticsRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).monitorDiagnosticSettingsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	monitoringId := parseMonitorDiagnosticId(d.Id())
+
+	resp, err := client.Get(ctx, monitoringId.ResourceID, monitoringId.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on Diagnostic Setting %s: %+v", monitoringId.Name, err)
+	}
+
+	d.Set("name", *resp.Name)
+
+	// ID of base resource is not returned by API, so we have to guess here
+	monitoringId = parseMonitorDiagnosticId(d.Id())
+	d.Set("target_resource_id", monitoringId.ResourceID)
+
+	if resp.StorageAccountID != nil {
+		d.Set("storage_account_id", *resp.StorageAccountID)
+	}
+
+	if resp.EventHubName != nil {
+		d.Set("event_hub_name", *resp.EventHubName)
+	}
+
+	if resp.EventHubAuthorizationRuleID != nil {
+		d.Set("event_hub_authorization_rule_id", *resp.EventHubAuthorizationRuleID)
+	}
+
+	if resp.WorkspaceID != nil {
+		d.Set("workspace_id", *resp.WorkspaceID)
+	}
+
+	d.Set("disabled_settings", flattenDisabledSettings(*resp.Metrics, *resp.Logs))
+	d.Set("retention_days", flattenRetentionDays(*resp.Metrics, *resp.Logs))
+
+	return nil
+}
+
+func flattenDisabledSettings(metricSettings []insights.MetricSettings, logSettings []insights.LogSettings) []interface{} {
+	disabledSettings := make([]interface{}, 0)
+
+	for _, setting := range metricSettings {
+		category := *setting.Category
+		if !*setting.Enabled {
+			disabledSettings = append(disabledSettings, category)
+		}
+	}
+
+	for _, setting := range logSettings {
+		category := *setting.Category
+		if !*setting.Enabled {
+			disabledSettings = append(disabledSettings, category)
+		}
+	}
+	return disabledSettings
+}
+
+func flattenRetentionDays(metricSettings []insights.MetricSettings, logSettings []insights.LogSettings) int32 {
+	returnSetting := int32(0)
+
+	for _, setting := range metricSettings {
+		if *setting.Enabled {
+			returnSetting = *setting.RetentionPolicy.Days
+		}
+	}
+
+	for _, setting := range logSettings {
+		if *setting.Enabled {
+			returnSetting = *setting.RetentionPolicy.Days
+		}
+	}
+
+	return returnSetting
+}
+
+func resourceArmMonitorDiagnosticsDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).monitorDiagnosticSettingsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	targetResoureId := d.Get("target_resource_id").(string)
```

Comment:
we need to parse the Resource ID out of the URI (since these values don't necessarily exist in the config/state, although they should) - I'll push a commit for this